doc/research/logo-detection/ann-benchmark.md

```diff
@@ -0,0 +1,88 @@
+# ANN Benchmark
+
+All logos are embedded through a computer vision model.
+In order to help logos annotation, the nearest neighbors of each logo must be easily accesible.
+A benchmark has been performed to determine the best way to perform a nearest neighbors search among the following HNSW indexes:
+
+- (faiss)[https://github.com/facebookresearch/faiss]
+- (redis)[https://redis.io/docs/stack/search/reference/vectors/]
+- (elasticsearch)[https://www.elastic.co/fr/blog/introducing-approximate-nearest-neighbor-search-in-elasticsearch-8-0]
+
+For each framework, multiple parameters were used to find the best index in our use case.
+
+## Our Use Case
+
+The data used to test the performance of the indexes was made of 4.371.343 vectors (each one issued from the CLIP vision model with a logo as input) of dimension 768 and type float32.
```

Comment:
We should add a paragraph to say we used the wrong embedding for the benchmark (size 768 instead of 512), but that it shouldn't change the result as there is only an additional projection. And that we use embedding of size 512 in production.